pylivetrader/backend/alpaca.py

```diff
@@ -590,27 +600,57 @@ def fetch(symbol):
             else:
                 df = self._api.get_barset(symbol,
                                           size,
-                                          start=_from.date().isoformat(),
-                                          end=to.date().isoformat(),
-                                          limit=limit).df[symbol]
+                                          start=_from.isoformat(),
+                                          end=to.isoformat()).df[symbol]
 
             # zipline -> right label
             # API result -> left label (beginning of bucket)
             if size == 'minute':
-                df.index += pd.Timedelta('1min')
+                df.index += one_min
 
                 if not df.empty:
                     # mask out bars outside market hours
                     mask = self._cal.minutes_in_range(
-                        df.index[0], df.index[-1],
+                        _from+one_min, to+one_min
                     ).tz_convert(NY)
                     df = df.reindex(mask)
 
             if limit is not None:
                 df = df.iloc[-limit:]
             return df
 
-        return parallelize(fetch)(symbols)
+        results = []
+        if self._use_polygon:
+            result = parallelize(fetch)(symbols)
+            for symbol, df in result.items:
+                df.columns = pd.MultiIndex.from_product([[symbol], [df.columns]])
+                results.append(df)
+        else:
+            # alpaca support get data of multiple assets(<200) at once
+            splitlen = 199
+            parts = []
+            for i in range(0, len(symbols), splitlen):
+                part = symbols[i:i + splitlen]
+                parts.append(part)
+            result = parallelize(fetch)(parts)
+            for v in result.values():
+                results.append(v)
+
+        # cause the pd.concat has performance problem, we use the np.append to concat the values
```

Comment:
this is the kind of comments/documentation I expect (I also learned that it works faster, so thanks)

Reply:
My mistake. the reason is I used the profile tool for performance testing, The profile tool itself will greatly increase the time consumption of the program. When i switch to Cprofile, The problem disappeared.